pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -397,7 +397,7 @@ func (m *ManagerImpl) UpdatePluginResources(node *schedulerframework.NodeInfo, a
 	defer m.mutex.Unlock()
 
 	// quick return if no pluginResources requested
-	if _, podRequireDevicePluginResource := m.podDevices[string(pod.UID)]; !podRequireDevicePluginResource {
+	if _, podRequireDevicePluginResource := m.podDevices.devs[string(pod.UID)]; !podRequireDevicePluginResource {
```

Comment:
This seems to be protected by the manager mutex, but would be nice to be a function call protected by the podDevices object :)

Reply:
yes, here and in other places manager's mutex.Lock could be avoided. But manager's mutex still protect m.allocatedDevices, m.healthyDevices, etc, and it could be also factored out to RWmutex or even to lock free structure.